testsuite/cluster-test/src/cluster_builder.rs

```diff
@@ -191,13 +217,45 @@ impl ClusterBuilder {
                 })
                 .collect();
             lsrs.append(&mut lsr_instances);
+        } else {
+            vault_nodes = vec![];
         }
 
+        let lsrs = try_join_all(lsrs).await?;
+        let vaults = try_join_all(vaults).await?;
+
         let validator_nodes = try_join_all((0..num_validators).map(|i| async move {
             let pod_name = validator_pod_name(i);
             self.cluster_swarm.allocate_node(&pod_name).await
         }))
         .await?;
+
+        let fullnode_nodes = try_join_all((0..num_validators).flat_map(move |validator_index| {
+            (0..num_fullnodes_per_validator).map(move |fullnode_index| async move {
+                let pod_name = fullnode_pod_name(validator_index, fullnode_index);
+                self.cluster_swarm.allocate_node(&pod_name).await
+            })
+        }))
+        .await?;
+
+        if !vault_nodes.is_empty() {
```

Comment:
Do you mind adding log before/after this code block, like 'generating genesis with management tool', 'done generating genesis'